### PR TITLE
[ISSUE-3812][Feature] Introduce token management e2e test

### LINKED ISSUE #3812: [Feature] Add Token Management e2e test cases
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Add Token Management e2e test cases:
- createToken
- deleteToken

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengkezhou1**
## What changes were proposed in this pull request

Issue Number: close #3812  

add basic token management test cases.

## Verifying this change

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (no)

### CODE REVIEW

**Comment by zhengkezhou1** [comment]
@SbloodyS PTAL

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@StreamPark(composeFiles = "docker/basic/docker-compose.yaml")
+public class TokenManagementTest {
```
Comment (SbloodyS) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java:36]: You should add it to https://github.com/apache/incubator-streampark/blob/cb57d06c2b4175da2f4fb86d090d0c96a07c38a0/.github/workflows/e2e.yml#L96-L105

**Review by SbloodyS — COMMENTED** [review]
Please add `testCopyToken` test case.

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  /**
+  * Users can only create one token. Once the user creates a token, it will not be found in the search layout.
+  * So type username which has been used in search layout will fail with frontend check. This method only used
+  * to help testCreateDuplicateToken testing frontend verify for and testDeleteToken skip frontend verify.
+  */
+  private void refreshPage() {
+      browser.navigate().refresh();
+  }
+}
```
Comment (zhengkezhou1) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java]: This method here is just to allow test cases to proceed.

**Comment by zhengkezhou1** [comment]
> Please add `testCopyToken` test case.

Copy token means copying the token to the clipboard. I think it should be added to the test case that repeatedly creates the token.

**Comment by SbloodyS** [comment]
> Copy token means copying the token to the clipboard. I think it should be added to the test case that repeatedly creates the token.

What I mean by that is, after copying to the clipboard, verify that the value of the clipboard is the same as the value of the page token text.

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+    @Test
+    @Order(20)
+    void testCopyToken() {
+        final TokenManagementPage tokenManagementPage = new TokenManagementPage(browser);
+        tokenManagementPage.copyToken(existUserName);
+
+        new WebDriverWait(browser, Duration.ofSeconds(10));
+        String clipboardContent = (String) ((JavascriptExecutor) browser).executeScript("return navigator.clipboard.readText();");
```
Comment (zhengkezhou1) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java]: @SbloodyS i am try a lot and found: `selenium` and `streampark` services running in docker, as remote service. I cannot visit the clipboard through **Toolkit** api under local env, another option is to use JS script to read the clipboard in  browser, the value of the board, but it failed for some reasons: maybe the page was not in focus or it was denied by the security policy.

**Comment by zhengkezhou1** [comment]
I try a lot and Its still hard to verify that token value equal in clipboard and page list.

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+        Awaitility.await()
+                .untilAsserted(
+                        () -> assertThat(tokenManagementPage.tokenList())
+                                .as("Token list should contain newly-created token")
+                                .extracting(WebElement::getText)
+                                .anyMatch(it -> it.contains(existUserName)));
+    }
+
+    // TODO: use browser clipboard replace local.
```
Comment (SbloodyS) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java]: You can try to use `String clipboardContent = (String) Toolkit.getDefaultToolkit().getSystemClipboard().getData(DataFlavor.stringFlavor);`

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                    .as("Token list should contain newly-created token")
+                    .extracting(WebElement::getText)
+                    .anyMatch(it -> it.contains(existUserName)));
+    }
+
+    @Test
+    @Order(20)
+    void testCopyToken() {
+        final TokenManagementPage tokenManagementPage = new TokenManagementPage(browser);
+        browser.navigate().refresh();
```
Comment (SbloodyS) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java]: We don't need to refresh page here.

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            .untilAsserted(
+                () -> assertThat(tokenManagementPage.errorMessageSearchLayout())
+                    .as("Please select a user").isNotNull());
+    }
+
+    @Test
+    @Order(40)
+    void testDeleteToken() {
+        final TokenManagementPage teamManagementPage = new TokenManagementPage(browser);
+        browser.navigate().refresh();
```
Comment (SbloodyS) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java]: Same here.

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                    .as("Clipboard should contain existing token.")
+                    .extracting(WebElement::getText)
+                    .anyMatch(it -> it.contains(token)));
+    }
+
+    @Test
+    @Order(30)
+    void testCreateDuplicateToken() {
+        final TokenManagementPage tokenManagementPage = new TokenManagementPage(browser);
+        browser.navigate().refresh();
```
Comment (SbloodyS) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java:96]: This still unaddressed.

Comment (zhengkezhou1) [reply streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java:96]: > This still unaddressed.

This place is necessary. Each user should only be able to create one token. At this time, the refresh will filter out the username that have already created tokens from the drop-down box.

Comment (zhengkezhou1) [reply streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java:96]: We can remove `browser.navigate().refresh();` after fixing issue https://github.com/apache/incubator-streampark/issues/3865

Comment (SbloodyS) [reply streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/TokenManagementTest.java:96]: I see.

**Review by SbloodyS — APPROVED** [review]
LGTM. @wolfboys PTAL

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**